internal/controller/datadogagent/controller_reconcile_v2_helpers.go

```diff
@@ -252,3 +259,61 @@ func generateNewStatusFromDDA(ddaStatus *datadoghqv2alpha1.DatadogAgentStatus) *
 	}
 	return status
 }
+
+// cleanupOldDCADeployments deletes DCA deployments when a DCA Deployment's name is changed using clusterAgent name override
+func (r *Reconciler) cleanupOldDCADeployments(ctx context.Context, logger logr.Logger, dda *datadoghqv2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, newStatus *datadoghqv2alpha1.DatadogAgentStatus) error {
+	matchLabels := client.MatchingLabels{
+		apicommon.AgentDeploymentComponentLabelKey: constants.DefaultClusterAgentResourceSuffix,
+		kubernetes.AppKubernetesManageByLabelKey:   "datadog-operator",
+		kubernetes.AppKubernetesPartOfLabelKey:     object.NewPartOfLabelValue(dda).String(),
+	}
+	deploymentName := component.GetDeploymentNameFromDatadogAgent(dda, &dda.Spec)
+	deploymentList := appsv1.DeploymentList{}
+	if err := r.client.List(ctx, &deploymentList, matchLabels); err != nil {
+		return err
+	}
+	for _, deployment := range deploymentList.Items {
+		if deploymentName != deployment.Name {
+			if _, err := r.cleanupV2ClusterAgent(ctx, logger, dda, &deployment, resourcesManager, newStatus); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// cleanupOldCCRDeployments deletes CCR deployments when a CCR Deployment's name is changed using clusterChecksRunner name override
+func (r *Reconciler) cleanupOldCCRDeployments(ctx context.Context, logger logr.Logger, dda *datadoghqv2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, newStatus *datadoghqv2alpha1.DatadogAgentStatus) error {
+	matchLabels := client.MatchingLabels{
+		apicommon.AgentDeploymentComponentLabelKey: constants.DefaultClusterChecksRunnerResourceSuffix,
+		kubernetes.AppKubernetesManageByLabelKey:   "datadog-operator",
+		kubernetes.AppKubernetesPartOfLabelKey:     object.NewPartOfLabelValue(dda).String(),
+	}
+	deploymentName := getDeploymentNameFromCCR(dda)
+	deploymentList := appsv1.DeploymentList{}
+	if err := r.client.List(ctx, &deploymentList, matchLabels); err != nil {
+		return err
+	}
+	for _, deployment := range deploymentList.Items {
+		if deploymentName != deployment.Name {
+			if _, err := r.cleanupV2ClusterChecksRunner(ctx, logger, dda, &deployment, resourcesManager, newStatus); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// getDeploymentNameFromCCR returns the expected CCR deployment name based on
+// the DDA name and clusterChecksRunner name override
+func getDeploymentNameFromCCR(dda *datadoghqv2alpha1.DatadogAgent) string {
+	deploymentName := clusterchecksrunner.GetClusterChecksRunnerName(dda)
+	if componentOverride, ok := dda.Spec.Override[datadoghqv2alpha1.ClusterChecksRunnerComponentName]; ok {
+		if componentOverride.Name != nil && *componentOverride.Name != "" {
+			deploymentName = *componentOverride.Name
+		}
+	}
+	return deploymentName
+}
```

Comment:
@ Timothée: common.GetDeploymentNameFromDatadogAgent gets DCA and does the same thing as this for CCR, only change is component + GetClusterAgentName vs GetClusterChecksRunnerName -> getComponentDeployNameFromDDA(ddaObject, ddaSpec, componentName) + getComponentName(dda, component)